account-keys/src/address_hash.rs

```diff
@@ -0,0 +1,44 @@
+//! A newtype representing a standard hash of a MobileCoin public address.
+//! This is used in certain memos, as a compact representation of the address.
+
+use crate::account_keys::PublicAddress;
+use core::convert::TryInto;
+use mc_crypto_digestible::{Digestible, MerlinTranscript};
+use subtle::{Choice, ConstantTimeEq};
+
+/// Represents a "standard" public address hash created using merlin,
+/// used in memos as a compact representation of a MobileCoin public address.
+/// This hash is collision resistant.
+#[derive(Debug, Clone, Eq, PartialEq, Ord, PartialOrd)]
+pub struct AddressHash([u8; 16]);
```

Comment:
I think it's possible I don't understand the request

We could have a second hash that's like

```
pub struct LongAddressHash([u8; 32]);
```

if you want that for some reason, but I'm not sure we actually need that?

I would rather just have two structs than do
```
ShortHash<'a>(&'a AddressHash);
```
because my goal here is ultimately to expose a typed API for the SDKs to bind to, and we cannot make use of a type like `ShortHash<'a>` as proposed for that.

Reply:
Actually, I guess I'm okay with calling this a `ShortAddressHash`, I will update

Reply:
done